src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestService.java

```diff
@@ -11,19 +11,22 @@ public interface HarvestService {
   /**
    * Harvest the given OAI endpoint with the given datasetId, data for the records and OAI-PMH data
    *
-   * @param datasetId The id of the dataset to be harvested
+   * @param datasetId              The id of the dataset to be harvested
    * @param recordDataEncapsulated The encapsulation of data to be used to harvest each record
-   * @param oaiHarvestData The object that encapsulate the necessary data for harvesting
+   * @param oaiHarvestData         The object that encapsulate the necessary data for harvesting
+   * @param stepSize
    */
-  void harvestOaiPmh(String datasetId, Record.RecordBuilder recordDataEncapsulated, OaiHarvestData oaiHarvestData);
+  void harvestOaiPmh(String datasetId, Record.RecordBuilder recordDataEncapsulated, OaiHarvestData oaiHarvestData,
```

Comment:
also here include `stepSize` with the `recordDataEncapsulated` using `DatasetMetadata` class.